client/src/main/scala/org/http4s/armeria/client/ArmeriaClient.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s
+package armeria
+package client
+
+import cats.effect.{Bracket, ConcurrentEffect, ContextShift, Resource}
+import cats.implicits._
+import com.linecorp.armeria.client.WebClient
+import com.linecorp.armeria.common.{
+  HttpData,
+  HttpHeaders,
+  HttpMethod,
+  HttpRequest,
+  HttpResponse,
+  RequestHeaders
+}
+import fs2.interop.reactivestreams._
+import fs2.{Chunk, Stream}
+import org.http4s.client.DefaultClient
+import org.reactivestreams.Publisher
+import org.http4s.internal.CollectionCompat.CollectionConverters._
+
+private[armeria] final class ArmeriaClient[F[_]] private[client] (
```

Comment:
You might be able to avoid the subtype by calling `Client.apply` with your `run` function.  Subtyping `DefaultClient` made it harder to fix something in Ember in a binary-compatible way.

Reply:
That is a good tip. I removed `DefaultClient` from the super type. 😀